server/api/main.py

```diff
@@ -677,6 +681,25 @@ def _push_terminal_run_notifications(db: server.api.db.base.DBInterface, db_sess
     _last_notification_push_time = now
 
 
+async def _abort_stale_runs(db_session, stale_runs: typing.List[dict]):
+    coroutines = []
+    for stale_run in stale_runs:
+        coroutine = fastapi.concurrency.run_in_threadpool(
+            server.api.crud.Runs().abort_run, db_session, **stale_run
+        )
+        coroutines.append(coroutine)
```

Comment:
use ` asyncio.Semaphore(` to ensure no more than X abort run requests executed in parallel